filebeat/registrar/registrar.go

```diff
@@ -259,9 +269,11 @@ func (r *Registrar) writeRegistry() error {
 	// Directly close file because of windows
 	f.Close()
```

Comment:
If an error occurs while encoding then the temp file is not closed. Add `f.Close()` to the `if` block above.